code/modules/biomatter_manipulation/solidifier.dm

```diff
@@ -0,0 +1,110 @@
+//Biomatter compressor
+//This machine converts liquid biomatter to solid one(sheets)
+//Working with this also required bio protection cloths
+
+#define BIOMATTER_PER_SHEET 		10
+#define CONTAINER_PIXEL_OFFSET 		6
+
+/obj/machinery/biomatter_solidifier
+	name = "Biomatter compressor"
+	icon = 'icons/obj/machines/simple_nt_machines.dmi'
+	icon_state = "solidifier"
+	density = TRUE
+	anchored = TRUE
+	use_power = 1
+	idle_power_usage = 5
+	active_power_usage = 300
+
+	circuit = /obj/item/weapon/circuitboard/neotheology/solidifier
+	var/active = FALSE
+	var/port_dir = NORTH
+	var/obj/structure/reagent_dispensers/biomatter/container
+
+
+/obj/machinery/biomatter_solidifier/update_icon()
+	if(active)
+		icon_state = "solidifier_on"
+	else
+		icon_state = initial(icon_state)
```

Comment:
It's better to use `icon_state = initial(icon_state) + "_on"` to keep ability to redefine icon state for child types.